tgui/packages/tgui/interfaces/LongRangeTeleporter.js

```diff
@@ -0,0 +1,49 @@
+import { useBackend } from '../backend';
+import { Button, LabeledList, Section } from '../components';
+import { Window } from '../layouts';
+
+export const LongRangeTeleporter = (props, context) => {
+  const { act, data } = useBackend(context);
+
+  const {
+    destinations,
+    receive_allowed,
+    send_allowed,
+    syndicate,
+  } = data;
+
+  return (
+    <Window
+      theme={syndicate ? 'syndicate' : 'ntos'}
+      width={390}
+      height={380}>
+      <Window.Content>
+        <Section title="Destinations">
+          <LabeledList>
+            {destinations.length ? destinations.map((d) => (
+              <LabeledList.Item label={d["destination"]} key={d["destination"]+d["ref"]}>
```

Comment:
```suggestion
              <LabeledList.Item label={d["destination"]} key={`${d["destination"]}}>
```
If "destination" is not unique then the user has no way of differentiating between the locations, so destination *should* be enough to uniquely identify the `Item` by.

If for some reason `ref` is needed as well, note that casting it to string will be `"[Object object]"` in cases where `ref` is null.

You could do the below instead:

```suggestion
              <LabeledList.Item label={d["destination"]} key={`${d["destination"]}${d["ref"] ? `-${d["ref"]}` : ''}}>
```

Reply:
Good catch, the implementation is myopic to the intended cases and ignorant of the possible user interaction aspect. ☹️

Reply:
@Azrun Updated the suggestion to be (I think) syntactically correct. I was missing a closing backtick in my initial version.